PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeShippingListener.kt

```diff
@@ -0,0 +1,36 @@
+package com.paypal.android.paypalnativepayments
+
+/**
+ * An optional listener to receive notifications if the user changes their shipping information.
+ */
+interface PayPalNativeShippingListener {
+
+    /**
+     * Notify when the users selected shipping address changes. Use [PayPalNativeShippingActions.approve]
+     * or [PayPalNativeShippingActions.reject] to approve or reject the newly selected shipping address.
+     * Optionally, if the order needs to be patched, call [PayPalNativeShippingActions.approve] once
+     * patching has completed successfully.
+     *
+     * @param actions actions to perform after a change in shipping address
+     * @param shippingAddress the user's most recently selected shipping address
+     */
+    fun onPayPalNativeShippingAddressChange(
+        actions: PayPalNativeShippingActions,
+        shippingAddress: PayPalNativeShippingAddress
+    )
+
+    /**
+     * Notify when the users selected a different shipping method. To reflect the newly selected
+     * shipping method in the paysheet, patch the order with operation 'replace', with all of the
```

Comment:
```suggestion
     * shipping method in the paysheet, patch the order on your server with operation 'replace', with all of the
```

A few suggestions to further clean up the docstring:
- Specify that they are required to call approve to continue the flow
- Specify that they should call approve or reject _after_ their server-side patching completes